### ISSUE TITLE
Unable to create Patient resource in transaction Bundle

### ISSUE THREAD

**Issue by lwarren01** [awslabs/fhir-works-on-aws-persistence-ddb#72]
When attempting to create a Patient resource within a transaction Bundle, the request fails returning a HTTP/400 response with the following error:
```
{
    "resourceType": "OperationOutcome",
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "diagnostics": "Cannot read property 'length' of undefined"
        }
    ]
}
```
The root cause of the error is the relative Organization reference provided in the managingOrganization field. The server is failing to parse the reference value as it is expecting an absolute URL instead.

If we change the request to include an absolute URL with the base URL for the server, the request completes successfully, creating the Patient resource. However, retrieving the Patient resource created returns an incorrect value in the managingOrganization field. Instead of the expected absolute URL, a relative value is returned with the wrong resource type, "Patient" instead of "Organization". The logical ID is correct though.

Here is the Bundle request we tested with:
```
{
    "resourceType": "Bundle",
    "type": "transaction",
    "entry": [
        {
            "resource": {
                "resourceType": "Patient",
                "active": true,
                "name": [
                    {
                        "family": "Smith",
                        "given": [
                            "John"
                        ]
                    }
                ],
                "gender": "male",
                "birthDate": "1996-09-24",
                "managingOrganization": {
                    "reference": "{fhirBaseUrl}/Organization/2f3df8d8-863e-47fb-a9b9-c634c8794fb9"
                },
                "address": [
                    {
                        "city": "Halifax",
                        "state": "NS",
                        "postalCode": "B4B2C3",
                        "country": "Canada"
                    }
                ]
            },
            "request": {
                "method": "POST",
                "url": "Patient"
            }
        }
    ]
}
```
And here is the Patient resource created:
```
{
    "birthDate": "1996-09-24",
    "meta": {
        "lastUpdated": "2021-05-01T12:25:02.256Z",
        "versionId": "1"
    },
    "managingOrganization": {
        "reference": "Patient/2f3df8d8-863e-47fb-a9b9-c634c8794fb9"
    },
    "address": [
        {
            "country": "Canada",
            "state": "NS",
            "city": "Halifax",
            "postalCode": "B4B2C3"
        }
    ],
    "name": [
        {
            "family": "Smith",
            "given": [
                "John"
            ]
        }
    ],
    "gender": "male",
    "active": true,
    "resourceType": "Patient",
    "id": "833164a5-7981-4e0e-a698-c6fb29340de5"
}
```
This issue was discovered while testing with v2.6.0.

**Comment by rsmayda** [comment]
I believe this PR will resolve this issue: https://github.com/awslabs/fhir-works-on-aws-routing/pull/74

We look to merge it in by early next week

**Comment by carvantes** [comment]
Hi, 

The fix has been released as part of `fhir-works-on-aws-routing@5.2.1`

I also suggest looking at the following comment that explains why relative references in a bundle without `fullUrl` is a "peculiar" use case: https://github.com/awslabs/fhir-works-on-aws-deployment/issues/305#issuecomment-830399502
